Improvements on ZetaNumbers branch

pheki opened the pull request.

pheki committed:
Fix example

pheki committed:
Try to fix github workflows

pheki committed:
Rename build-util to vitasdk-sys-build-util

pheki committed:
Install LLVM and Clang for build, clippy and doc

pheki committed:
Change Clang version to 16

pheki committed:
Allow formatting to fail on build.rs as the user may not have rustfmt

zetanumbers commented on build-util/Cargo.toml:
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "build-util"
```
~~Could i ask why rename it? It's not meant for publishing since it works only inside the repo, which has it's code anyway. I was doing something like [cargo-xtask](https://github.com/matklad/cargo-xtask) approach, or more precisely like [rust-analyzer does](https://github.com/rust-lang/rust-analyzer/blob/609bdbc4ddcd40d480730876595c0b7f4a58ffcc/xtask/Cargo.toml). On that note, ra seems to just set `publish = false`.~~

zetanumbers replied:
It's `vitasdk-sys` dependency, which we have to publish, you're right.

zetanumbers committed:
Fix error message reffering to build-util after rename

zetanumbers committed:
Do not use VITASDK env var cause we don't need it

zetanumbers committed:
Use CARGO_MANIFEST_DIR relative paths

pheki committed:
Add submodules to clippy and doc

pheki committed:
Actually add submodules for doc...

zetanumbers committed:
Update lib.rs from upstream

pheki committed:
Go back to signed char

pheki committed:
Upload docs on action

pheki committed:
Stop caching `target/`

pheki commented on build.rs:
```diff
@@ -1,32 +1,18 @@
-
-    let lib = sysroot.join("lib");
-    assert!(lib.exists(), "VITASDK's `lib` directory does not exist");
-    println!("cargo:rustc-link-search=native={lib}");
```
@ZetaNumbers removing `cargo:rustc-link-search=native={lib}` broke it for me and CI, as it's used to find the stubs when linking... Is it ok to revert commit https://github.com/vita-rust/vitasdk-sys/pull/11/commits/17653562ac19a53014d865c938906ffafd290ddb?

zetanumbers committed:
Revert 2 commits

This reverts commits 17653562ac19a53014d865c938906ffafd290ddb
and dc2c1612f1439d9509c9928a06a32806b89ca64d.

pheki committed:
Remove dbg!() from build.rs

pheki committed:
Build docs for armv7-sony-vita-newlibeabihf

zetanumbers committed:
fix ci

pheki committed:
Fix docs CI

pheki committed:
Remove VITASDK from docs action

pheki committed:
Warn on rustfmt failure

pheki committed:
Add env to docs CI

pheki committed:
Pass --no-deps to docs CI

pheki committed:
Increase upload docs timeout

pheki committed:
Remove strange comment

zetanumbers committed:
Add myself to the authors

zetanumbers committed:
Remove links field

> This manifest states that the package links to the libfoo native library. When using the links key, the package must have a build script, and the build script should use the rustc-link-lib instruction to link the library.

https://doc.rust-lang.org/cargo/reference/build-scripts.html#the-links-manifest-key

zetanumbers committed:
Add `--fail-if-any` flag to build-util

zetanumbers committed:
ignore empty arguments in build-util

zetanumbers committed:
Add `Check missing libs` CI workflow

zetanumbers committed:
run build workflow on push only on main branch

pheki committed:
Add ZetaNumbers as author of vitasdk-sys-build-util

pheki committed:
Update READMEs

pheki committed:
Use environment variable for LLVM version on build workflow

pheki committed:
Specify vitasdk-sys-build-util version for crates.io

pheki committed:
Bump crate version

pheki committed:
Update CHANGELOG.md

pheki committed:
Increase timeout for doc upload
